### ISSUE TITLE
Will we have speaker-related information included in the data package?

### ISSUE THREAD

**Issue by underdogliu** [SpeechColab/GigaSpeech#61]
Hello and great job on the dataset! Finally we have a 10k-hour dataset with internal strong baselines that can benefit the whole community. 

As a speech processing researcher I would definitely take advantage of it. But also as my focus is mainly on speaker-related stuff, I would like to ask if speaker-related information will be associated. It does not have to be very detailed if speakers' privacy is concerned. It can be as simple as a metadata which can be used to generated something like `spk2utt` in kaldi, for instance.

For more questions, we can keep discussing in this thread. Thanks for being understanding and supportive!

**Comment by chenguoguo** [comment]
Yes adding speaker labels is one of the directions that we would love to explore, and we even have this entry "speaker" in the GigaSpeech.json file. Currently that entry is not set.

Some of the original transcription comes with speaker names, so it should be doable. Bottom line is that we can create a subset with decent speaker labels.

But before we start the implementation, we also would love to hear from people on what is the best practice. Once we sort that out, we can add speaker information to GigaSpeech.json and have a newer release. Ideas/helps are welcome!

**Comment by underdogliu** [comment]
@chenguoguo Thanks a lot! As a speaker verification guy, I can definitely collaborate and provide some information and practical effort. 

So in general, I believe speaker IDs shall consists of essential information such as unique speaker names (or just IDs, if privacy is concerned), gender, age and recording environment. All of those can be helpful for speaker verification research. Format depends on toolkits that are used. I think we mainly focus on Kaldi, espnet and wenet? 

But we can talk more. I am familar with Kaldi but not as familiar speaking of the other two.

**Comment by chenguoguo** [comment]
I think it's fine if we start this with Kaldi. The limitation is really on the original data source. I believe we will only be able to use part of the data. Feel free to send me a private email so that I can explain the limitations to you.

**Comment by underdogliu** [comment]
> I think it's fine if we start this with Kaldi. The limitation is really on the original data source. I believe we will only be able to use part of the data. Feel free to send me a private email so that I can explain the limitations to you.

Yep I sent you private email already while not sure if it's the right email address. If you receive the mail you can close this issue. Otherwise please tell me your work email address so I can send it to the correct one. Thanks!

**Comment by desh2608** [comment]
Is there any update on adding speaker id information to Gigaspeech? Would be great even if just the DEV/TEST recordings can be annotated for starters :)

**Comment by underdogliu** [comment]
@desh2608 Yeah I was discussing with @chenguoguo like two years ago about this. But then I found such audio is a bit hard to annotate and preparing trials and I have got something else to do so this has been pended for two years. I am very sorry about this 😓 

One way I can tell for annotating the data is to use pseudo-anonymised IDs via similarity measure, using a very strong speaker embedding extractor. But of course, in such case, there will be a lot of wrong IDs.

I guess the perfectly right way is to manually annotate stuff. But this takes quite huge effort.

Here is a Google Doc on the issue I was noting couple years ago. We can discuss about it: [link](https://docs.google.com/document/d/11QJ1QqeMWLw6cBHi_mWGD8v9xjjuRlfhKhSmXaWHtHY/edit?usp=sharing)
